packages/quantic/force-app/main/default/lwc/quanticTimeframeFacet/e2e/quanticTimeframeFacet.e2e.ts

```diff
@@ -0,0 +1,159 @@
+          baseFacet.waitForFacetSelectUaAnalytics(expectedFacetData);
+        let searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+
+        await facetSelectUaRequest;
+        let searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetSelect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+
+        const clearAllUaRequest = baseFacet.waitForFacetClearAllUaAnalytics({
+          facetId: exampleField,
+          facetField: exampleField,
+        });
+        searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnClearSelectionButton();
+
+        await clearAllUaRequest;
+        searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetDeselect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetDeselect.actionCause).toEqual('facetClearAll');
+      });
+    });
+
+    test.describe('when typing in the facet search box input', () => {
+      test('should fetch facet values according to the query typed', async ({
+        baseFacet,
+        facet,
+      }) => {
+        const exampleStart = '2025-01-01';
+        const exampleEnd = '2025-01-02';
+        const expectedFacetData = {
+          facetId: `${exampleField}_input`,
+          facetField: exampleField,
+          facetValue: `${exampleStart.replaceAll('-', '/')}@00:00:00..${exampleEnd.replaceAll('-', '/')}@23:59:59`,
+        };
+
+        const facetSelectUaRequest =
+          baseFacet.waitForFacetSelectUaAnalytics(expectedFacetData);
+        const searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.fillFilterStartInput(exampleStart);
+        await facet.fillFilterEndInput(exampleEnd);
+        await facet.clickOnFilterApplyButton();
+
+        await facetSelectUaRequest;
+        const searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetSelect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+        if (useCase.value === useCaseEnum.search) {
+          expect(analyticsForFacetSelect.customData).toEqual(
+            expect.objectContaining(expectedFacetData)
+          );
+        }
+      });
+    });
+
+    if (useCase.value === useCaseEnum.search) {
+      test.describe('with a selected value in the URL', () => {
+        const selectedIndex = 0;
+        const date = new Date();
+        const todayDate = date.toISOString().slice(0, 10);
+        date.setDate(date.getDate() - 1);
+        const yesterdayDate = date.toISOString().slice(0, 10);
+        test.use({
+          urlHash: `df-${exampleField}=${yesterdayDate.replaceAll('-', '/')}@00:00:00..${todayDate.replaceAll('-', '/')}@23:59:59`,
```

Comment:
Do you think it's worth making a function to convert a start and end to the proper facet state/url format?

Reply:
Personally,  I don't really feel the need that much, first because this code is not repeated a lot, just written twice, and second because personally it seems simpler for to just see directly what's going on on the test instead of needing to scroll up to this utility function.